lib/improver/nbhood/nbhood.py

```diff
@@ -391,3 +378,165 @@ def process(self, cube):
+            ens_factor=1.0, weighted_mode=True):
+        """
+        Create a neighbourhood processing plugin that applies a smoothing
+        to points in a cube.
+
+        Parameters
+        ----------
+        neighbourhood_method : str
+            Name of the neighbourhood method to use. Options: 'circular',
+            'square'.
+        radii : float or List (if defining lead times)
+            The radii in metres of the neighbourhood to apply.
+            Rounded up to convert into integer number of grid
+            points east and north, based on the characteristic spacing
+            at the zero indices of the cube projection-x and y coords.
+        lead_times : None or List (optional)
+            List of lead times or forecast periods, at which the radii
+            within 'radii' are defined. The lead times are expected
+            in hours.
+        ens_factor : float (optional)
+            The factor with which to adjust the neighbourhood size
+            for more than one ensemble member.
+            If ens_factor = 1.0 this essentially conserves ensemble
+            members if every grid square is considered to be the
+            equivalent of an ensemble member.
+            Optional, defaults to 1.0
+        weighted_mode : boolean (optional)
+            If True, use a circle for neighbourhood kernel with
+            weighting decreasing with radius.
+            If False, use a circle with constant weighting.
+
+        """
+        super(NeighbourhoodProcessing, self).__init__(
+            neighbourhood_method, radii, lead_times=lead_times,
+            ens_factor=ens_factor)
+
+        methods = {
+            "circular": CircularNeighbourhood,
+            "square": SquareNeighbourhood}
+        try:
+            method = methods[neighbourhood_method]
+            self.neighbourhood_method = method(weighted_mode)
+        except KeyError:
+            msg = ("The neighbourhood_method requested: {} is not a "
+                   "supported method. Please choose from: {}".format(
+                       neighbourhood_method, methods.keys()))
+            raise KeyError(msg)
+
+        self.weighted_mode = bool(weighted_mode)
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<{}; weighted_mode: {}>')
+        return result.format(
+            super(NeighbourhoodProcessing, self).__repr__(),
+            self.weighted_mode)
+
+    def process(self, cube):
+        """
+        Apply probabilistic neighbourhood processing.
+
+        Parameters
+        ----------
+        cube : Iris.cube.Cube
+            Cube to apply a neighbourhood processing method to, in order to
+            generate a smoother field.
+
+        Returns
+        -------
+        Iris.cube.Cube
+            Cube after applying a neighbourhood processing method, so that the
+            resulting field is smoothed.
+        """
+        return self.run(cube)
```

Comment:
See comment in `GeneratePercentilesFromANeighbourhood`

Reply:
Good point. I've removed this method.